api/src/main/java/org/openmrs/module/reporting/report/manager/ReportManagerUtil.java

```diff
@@ -159,6 +160,25 @@ public static ReportDesign createExcelTemplateDesign(String reportDesignUuid, Re
 		return design;
 	}
 
+	/**
+	 * @return a new ReportDesign for a JSON template, using a file on the classpath as the template
+	 */
+	public static ReportDesign createJSONTemplateDesign(String reportDesignUuid, ReportDefinition reportDefinition, String resourcePath) {
```

Comment:
@dilanthas, this line does exactly what you asked for

Reply:
Yes.. In your code from line 167-171 does the same thing as in this method. You could use this method to create ReportDesign object in line 167 and remove line 168-171 ?

Reply:
I don't think i am following, the createJSONTemplateDesign method works just as others like 
createExcelTemplateDesign, createCsvReportDesign work. for every format there is a template design method

Reply:
Sorry for being not clear.. Let me try to explain. in line 167 you are creating a ReportDesign design object with some attributes. You also have a method called createJSONReportDesign() which does the same thing if im not mistaken. So my question is can you do `ReportDesign design = createJSONReportDesign();` in line 167 ?

Reply:
I have implemented the changes as suggested, turns out the code is cleaner than before, thank you for the feed back